views/templates/hook/express-checkout.tpl

```diff
@@ -0,0 +1,174 @@
+                        if (spinner) {
+                            spinner.overlay.appendTo('body');
+                            spinner.css.appendTo('head');
+                        }
+
+                        jQuery.ajax({
+                            url: '{$EXPRESS_CONTROLLER|escape:'htmlall':'UTF-8'}',
+                            method: 'POST',
+                            data: {
+                                action: 'complete_order',
+                                token: event.data.orderToken,
+                                urlToken: AfterPay.currentExpressUrlToken,
+                                key: "{$SECURE_KEY|escape:'htmlall':'UTF-8'}"
+                            },
+                            success: function(ajax_data){
+                                console.log("success", ajax_data);
+                                if (ajax_data.success === true) {
+                                    jQuery('.btn-afterpay_express').prop('disabled', false);
+                                    window.location.href = ajax_data.url;
+                                    if (spinner) {
+                                        spinner.overlay.remove();
+                                        spinner.css.remove();
+                                    }
+                                } else {
+                                    alert("error: " + ajax_data.message);
+                                    window.location.href = ajax_data.url;
+                                }
+
+                            },
+                            error: function(request, statusText, errorThrown) {
+                                jQuery('.btn-afterpay_express').prop('disabled', false);
+                                console.log('Something went wrong. Please try again later.');
+                                //window.location.href = data.url;
+                                if (spinner) {
+                                    spinner.overlay.remove();
+                                    spinner.css.remove();
+                                }
+                            }
+                        });
+                    } else {
+                        jQuery('.btn-afterpay_express').prop('disabled', false);
+                    }
+                }
+            },
+        });
+    }
+    document.addEventListener('readystatechange', event => {
+        if (event.target.readyState === "interactive" &&
+            typeof window.AfterPay !== 'undefined' &&
+            typeof AfterPay.initializeForPopup != 'undefined'
+        ) {
+            initAfterPayExpress();
+        }
+
+        // Double check
+        if (event.target.readyState === "complete" &&
+            typeof window.AfterPay !== 'undefined' &&
+            typeof AfterPay.initializeForPopup != 'undefined' &&
+            document.getElementById('afterpay_express_button').disabled == true
+        ) {
+            initAfterPayExpress();
+        }
+    });
```

Comment:
Just some suggestions.
[This](https://github.com/AfterpayTouch/clearpay-woocommerce/blob/master/js/clearpay_express.js#L5) was done by Scott, he is a full stack eng and excels at front-end.

The button is disabled for a few seconds because the call stack has to loop through all the ready states when the real requirement is the button element and the Afterpay object. Therefore, we could possibly remove further checks and if we need to do so we can start checking from [events](https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event) `DOMContentLoaded ` or `load`

I think passing [jQuery as a param](https://github.com/AfterpayTouch/clearpay-woocommerce/blob/master/js/clearpay_express.js#L14) in an [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) like Scott did is safer and allows avoiding having to loop through the documents ready states.

Additionally, the Ajax methods depend on jQuery an exception would be thrown if jQuery is not in the window object.